src/page.ts

```diff
@@ -0,0 +1,308 @@
+    // deno-lint-ignore no-explicit-any
+  ): Promise<any> {
+    if (typeof pageCommand === "string") {
+      const result = await this.#protocol.sendWebSocketMessage<
+        Protocol.Runtime.EvaluateRequest,
+        Protocol.Runtime.EvaluateResponse
+      >("Runtime.evaluate", {
+        expression: pageCommand,
+        includeCommandLineAPI: true, // supports things like $x
+      });
+      await this.#checkForErrorResult(result, pageCommand);
+      return result.result.value;
+    }
+
+    if (typeof pageCommand === "function") {
+      const { executionContextId } = await this.#protocol.sendWebSocketMessage<
+        Protocol.Page.CreateIsolatedWorldRequest,
+        Protocol.Page.CreateIsolatedWorldResponse
+      >(
+        "Page.createIsolatedWorld",
+        {
+          frameId: this.#protocol.frame_id,
+        },
+      );
+
+      const res = await this.#protocol.sendWebSocketMessage<
+        Protocol.Runtime.CallFunctionOnRequest,
+        Protocol.Runtime.CallFunctionOnResponse
+      >(
+        "Runtime.callFunctionOn",
+        {
+          functionDeclaration: pageCommand.toString(),
+          executionContextId: executionContextId,
+          returnByValue: true,
+          awaitPromise: true,
+          userGesture: true,
+        },
+      );
+      await this.#checkForErrorResult(res, pageCommand.toString());
+      return res.result.value;
+    }
+  }
+
+  /**
+   * Wait for the page to change. Can be used with `click()` if clicking a button or anchor tag that redirects the user
+   */
+  async waitForPageChange(): Promise<void> {
+    const method = "Page.loadEventFired";
+    this.#protocol.notification_resolvables.set(method, deferred());
+    const notificationPromise = this.#protocol.notification_resolvables.get(
+      method,
+    );
+    await notificationPromise;
+    this.#protocol.notification_resolvables.delete(method);
+  }
+
+  /**
+   * Check if the given text exists on the dom
+   *
+   * @param text - The text to check for
+   */
+  async assertSee(text: string): Promise<void> {
+    const command = `document.body.innerText.includes('${text}')`;
+    const exists = await this.evaluate(command);
+    if (exists !== true) { // We know it's going to fail, so before an assertion error is thrown, cleanupup
+      await this.#protocol.done();
+    }
+    assertEquals(exists, true);
+  }
+
+  // deno-lint-ignore require-await
+  async #$x(_selector: string) {
+    throw new Error("Client#$x not impelemented");
+    // todo check the element exists first
+    //return new Element('$x', selector, this)
+  }
+
+  /**
+   * Representation of the Browsers `document.querySelector`
+   *
+   * @param selector - The selector for the element
+   *
+   * @returns An element class, allowing you to action upon that element
+   */
+  async querySelector(selector: string) {
+    const result = await this.evaluate(
+      `document.querySelector('${selector}')`,
+    );
+    if (result === null) {
```

Comment:
It's tried and tested, and a falsey value is returned even if the selector is true i think, so a `!value` check would always be hit, i wrote it this way because i knew that if the element didn't exist, `null` would be returned, whereas if the elem did, `null` **wouldnt** be returned, but i see your concern

Reply:
daaang. nuance. alright. resolving